spec/requests/users_spec.rb

```diff
@@ -1,10 +1,33 @@
 require 'rails_helper'
 
 RSpec.describe 'Users', type: :request do
-  describe 'GET /index' do
-    it 'returns http success' do
+  describe 'Users' do
+    it 'returns http success, Correct Body , Correct Template' do
       get '/users/index'
       expect(response).to have_http_status(:success)
+      expect(response.body).to include('User List Here')
+      expect(response).to render_template(:index)
+    end
+
+    it 'returns http success, Correct Body , Correct Template' do
+      get '/users/id/posts'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include('HERE USER POSTS')
+      expect(response).to render_template(:posts)
+    end
+
+    it 'returns http success, Correct Body , Correct Template' do
+      get '/users/id/posts/id'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include('HERE A SINGLE POST')
+      expect(response).to render_template(:post)
+    end
```

Comment:
Kindly, move test for post in its file `posts_spec.rb` to keep your code base look professional. That way we'll have two files inside `spec/requests` one for each component user, and post.

Reply:
@Mre55 ty for the comment, I only have one controller `users` that's why, do you suggest creating a separate controller `posts` as well with the test file? or not necessary